examples/30_extended/flow_id_tutorial.py

```diff
@@ -0,0 +1,69 @@
+analysis.
+"""
+
+####################################################################################################
+import sklearn.tree
+
+import openml
+
+clf = sklearn.tree.DecisionTreeClassifier()
+
+####################################################################################################
+# 1. Obtaining a flow given a classifier
+# ======================================
+#
+
+flow = openml.extensions.get_extension_by_model(clf).model_to_flow(clf).publish()
+flow_id = flow.flow_id
+print(flow_id)
+
+####################################################################################################
+# This piece of code is rather involved. First, it retrieves an
+# :class:`~openml.extensions.Extension` which is registered and can handle the given model,
+# in our case it is :class:`openml.extensions.sklearn.SklearnExtension`. Second, the extension
+# converts the classifier into an instance of :class:`openml.flow.OpenMLFlow`. Third and finally,
+# the publish method checks whether the current flow is already present on OpenML. If not,
+# it uploads the flow, otherwise, it updates the current instance with all information computed
+# by the server (which is obviously also done when uploading/publishing a flow).
+#
+# To simplify the usage we have created a helper function which automates all these steps:
+
+flow_id = openml.flows.get_flow_id(model=clf)
+print(flow_id)
+
+####################################################################################################
+# 2. Obtaining a flow given its name
+# ==================================
+# The schema of a flow is given in XSD (`here
+# <https://github.com/openml/OpenML/blob/master/openml_OS/views/pages/api_new/v1/xsd/openml.implementation.upload.xsd>`_).  # noqa E501
+# Only two fields are required, a unique name, and an external version. While it should be pretty
+# obvious why we need a name, the need for the additional external version information might not
+# be immediately clear. However, this information is very important as it allows to have multiple
+# flows with the same name for different versions of a software. This might be necessary if an
+# algorithm or implementation introduces, renames or drop hyperparameters over time.
+
+flow = openml.extensions.get_extension_by_model(clf).model_to_flow(clf)
+print(flow.name, flow.external_version)
+
+####################################################################################################
+# The name and external version are automatically added to a flow when constructing it from a
+# model. We can then use them to retrieve the flow id as follows:
+
+flow_id = openml.flows.flow_exists(name=flow.name, external_version=flow.external_version)
+print(flow_id)
+
+####################################################################################################
+# We can also retrieve all flows for a given name:
+flow_ids = openml.flows.get_flow_id(name=flow.name, exact_version=False)
```

Comment:
I understand that you want to emphasize the fact that we get multiple flow ids and there is no `exact_version` when given a flow name. However, in the code, the exact_version is always set to `False`, no matter what the user gives.

Reply:
No, the user can give a model, and then `exact_version` makes a difference in the code (code = function `get_flow_id`). Are you talking about the example or the package code?

Reply:
The example, `exact_version` does not add or remove anything when giving a flow name as above. It is always replace by `False` in the `get_flow_id` method.
It was a remark, I understand that this way it is clearer to a new user.